kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/DefaultKroxyliciousTester.java

```diff
@@ -54,7 +58,14 @@ private KroxyliciousClients clients() {
     }
 
     private KroxyliciousClients clients(String virtualCluster) {
-        return new KroxyliciousClients(KroxyliciousConfigUtils.bootstrapServersFor(virtualCluster, kroxyliciousConfig));
+        if (clients.containsKey(virtualCluster)) {
+            return clients.get(virtualCluster);
+        }
+        else {
+            KroxyliciousClients client = new KroxyliciousClients(KroxyliciousConfigUtils.bootstrapServersFor(virtualCluster, kroxyliciousConfig));
+            clients.put(virtualCluster, client);
+            return client;
+        }
```

Comment:
Nit: as we are single threaded but using computeIfAbsent would make this an atomic operation